upup/pkg/kutil/rollingupdate_cluster.go

```diff
@@ -269,9 +270,42 @@ func buildCloudInstanceGroup(ig *api.InstanceGroup, g *autoscaling.Group, nodeMa
 	return n
 }
 
-func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, k8sClient *k8s_clientset.Clientset) error {
+func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, headroom int64, k8sClient *k8s_clientset.Clientset) error {
 	c := cloud.(awsup.AWSCloud)
 
+	if headroom > 0 {
+		describeParams := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{aws.String(n.ASGName)},
+			MaxRecords: aws.Int64(1),
+		}
+
+		out, err := c.Autoscaling().DescribeAutoScalingGroups(describeParams)
+		if err != nil {
+			return fmt.Errorf("Error adding headroom to ASG %s: %v", n.ASGName, err)
+		}
+
+		currentCapacity := aws.Int64Value(out.AutoScalingGroups[0].DesiredCapacity)
+		desiredCapacity := currentCapacity + headroom
+
+		if desiredCapacity > *out.AutoScalingGroups[0].MaxSize {
+			return fmt.Errorf("Adding headroom exceeds ASG max, increase ASG max to continue.")
```

Comment:
From my point of view:
  * cluster-autoscaler should be able to read max, min values from ASG to have a single source of truth or maybe write its if you specify the parameters due it will be a hardlimit in anycase
  * Use this values from ASG to add capacity in a rolling-update event. Since cluster-autoscaler has a grace period of 10 minutes (by default) to delete a node once it detects its not needed I think the process will not conflict. Or in specific cases this period time could be aumented